meteor/client/ui/SegmentTimeline/SegmentTimelinePart.tsx

```diff
@@ -428,533 +454,587 @@ const CARRIAGE_RETURN_ICON = (
-										duration={SegmentTimelinePart0.getPartDuration(this.props, this.state.liveDuration)}
-										expectedDuration={SegmentTimelinePart0.getPartExpectedDuration(this.props)}
-										isLiveLine={this.props.playlist.currentPartInstanceId === part.instance._id}
-										isNextLine={this.props.playlist.nextPartInstanceId === part.instance._id}
-										timeScale={this.props.timeScale}
-										autoNextPart={this.props.autoNextPart}
-										liveLinePadding={SegmentTimelinePart0.getLiveLineTimePadding(this.props.timeScale)}
-										indexOffset={currentIndex}
-									/>
-								)
-							}
-						})
-				}
+	getLayerStyle() {
+		// this.props.part.expectedDuration ||
+		const actualPartDuration = SegmentTimelinePartClass.getPartDuration(this.props, this.state.liveDuration)
+		const partDuration = this.props.cropDuration
+			? Math.min(this.props.cropDuration, actualPartDuration)
+			: actualPartDuration
+		if (this.props.relative) {
+			return {
+				width: ((partDuration / (this.props.totalSegmentDuration || 1)) * 100).toString() + '%',
+				// width: (Math.max(this.state.liveDuration, this.props.part.duration || this.props.part.expectedDuration || 3000) / (this.props.totalSegmentDuration || 1) * 100).toString() + '%',
+				willChange: this.state.isLive ? 'width' : undefined,
 			}
-
-			getFutureShadeStyle = () => {
-				return {
-					width:
-						Math.min(
-							Math.max(
-								0,
-								(this.props.livePosition || 0) +
-									SegmentTimelinePart0.getLiveLineTimePadding(this.props.timeScale) -
-									(this.props.part.instance.part.expectedDuration || this.props.part.renderedDuration || 0)
-							),
-							SegmentTimelinePart0.getLiveLineTimePadding(this.props.timeScale)
-						) *
-							this.props.timeScale +
-						'px',
-				}
+		} else {
+			return {
+				minWidth: Math.floor(partDuration * this.props.timeScale).toString() + 'px',
+				// minWidth: (Math.max(this.state.liveDuration, this.props.part.duration || this.props.part.expectedDuration || 3000) * this.props.timeScale).toString() + 'px',
+				willChange: this.state.isLive ? 'minWidth' : undefined,
 			}
+		}
+	}
 
-			static convertHexToRgba(hexColor: string): { red: number; green: number; blue: number } | undefined {
-				if (hexColor.substr(0, 1) !== '#') return
-				if (hexColor.length !== 7) return
+	static getPartExpectedDuration(props: WithTiming<IProps>): number {
+		return (
+			props.part.instance.timings?.duration ||
+			(props.timingDurations.partDisplayDurations &&
+				props.timingDurations.partDisplayDurations[unprotectString(props.part.instance.part._id)]) ||
+			props.part.renderedDuration ||
+			0
+		)
+	}
 
-				const red = parseInt(hexColor.substr(1, 2), 16)
-				const green = parseInt(hexColor.substr(3, 2), 16)
-				const blue = parseInt(hexColor.substr(5, 2), 16)
+	static getPartDuration(props: WithTiming<IProps>, liveDuration: number): number {
+		// const part = this.props.part
```

Comment:
Remove dead code or add a comment saying why it should be preserved

Reply:
Should be removed.